pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -89,38 +130,117 @@ function make_PyTerminal(app: PyScriptApp) {
             this.outElem.className = 'py-terminal';
             this.appendChild(this.outElem);
 
-            if (this.isAuto()) {
-                this.classList.add('py-terminal-hidden');
-                this.autoShowOnNextLine = true;
-            } else {
-                this.autoShowOnNextLine = false;
-            }
+            this.setupPosition(app);
+        }
 
+        // implementation of the Stdio interface
+        stdout_writeline(msg: string) {
+            this.outElem.innerText += msg + '\n';
             if (this.isDocked()) {
-                this.classList.add('py-terminal-docked');
+                this.scrollTop = this.scrollHeight;
+            }
+            if (this.autoShowOnNextLine) {
+                this.classList.remove('py-terminal-hidden');
+                this.autoShowOnNextLine = false;
             }
+        }
 
-            logger.info('Registering stdio listener');
-            app.registerStdioListener(this);
+        stderr_writeline(msg: string) {
+            this.stdout_writeline(msg);
         }
+        // end of the Stdio interface
+    }
 
-        isAuto() {
-            return this.hasAttribute('auto');
+    return PyTerminalPre;
+}
+
+declare const Terminal: typeof TerminalType;
+
+function make_PyTerminal_xterm(app: PyScriptApp) {
+    /** The <py-terminal> custom element, which automatically register a stdio
+     *  listener to capture and display stdout/stderr
+     */
+    class PyTerminalXterm extends PyTerminalBaseClass {
+        outElem: HTMLDivElement;
+        moduleResolved: boolean;
+        xtermReadyPromise: Promise<TerminalType>;
+        xterm: TerminalType;
+        cachedStdOut: Array<string>;
+        cachedStdErr: Array<string>;
+
+        constructor() {
+            super();
+            this.moduleResolved = false;
+            this.cachedStdOut = [];
+            this.cachedStdErr = [];
+
+            //Required to make xterm appear properly
+            this.style.width = '100%';
+            this.style.height = '100%';
         }
 
-        isDocked() {
-            return this.hasAttribute('docked');
+        async connectedCallback() {
+            this.outElem = document.createElement('div');
```

Comment:
I wonder if we could guard multiple initialization of the same terminal in case some 3rd party library moves the node around as addressed in this PR for `py-script` components: https://github.com/pyscript/pyscript/pull/1433

if not relevant for now, feel free to ignore this comment.

Reply:
Saw that PR was merged - is that a good general-purpose strategy for preventing multiple initialization? I.e. keep a list of initalized elements, and check whether the current element is in it before starting to initialize it?